src/components/Accounts/Actions/ActionModal.js

```diff
@@ -0,0 +1,513 @@
+import _ from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import {
+  Field,
+  reduxForm,
+  change,
+} from 'redux-form';
+import SafeHTMLMessage from '@folio/react-intl-safe-html';
+import {
+  Row,
+  Col,
+  Button,
+  TextArea,
+  Modal,
+  TextField,
+  Checkbox,
+  Select,
+} from '@folio/stripes/components';
+
+import { calculateSelectedAmount } from '../accountFunctions';
+import css from './PayWaive.css';
+
+const validate = (values, props) => {
+  const {
+    accounts,
+    action,
+    commentRequired
+  } = props;
+
+  const selected = calculateSelectedAmount(accounts);
+  const errors = {};
+
+  if (!values.amount) {
+    errors.amount = <FormattedMessage id="ui-users.accounts.error.field" />;
+  }
+  if (values.amount <= 0) {
+    errors.amount = <FormattedMessage id={`ui-users.accounts.${action}.error.amount`} />;
+  }
+  if (!values.method) {
+    errors.method = <FormattedMessage id="ui-users.accounts.error.select" />;
+  }
+  if (commentRequired && !values.comment) {
+    errors.comment = <FormattedMessage id="ui-users.accounts.error.comment" />;
+  }
+  if (parseFloat(values.amount) > parseFloat(selected)) {
+    errors.amount = <FormattedMessage id={`ui-users.accounts.${action}.error.exceeds`} />;
+  }
+
+  return errors;
+};
+
+const onChange = (values, dispatch, props, prevValues) => {
+  if (values.ownerId !== prevValues.ownerId) {
+    dispatch(change('payment-many-modal', 'method', null));
+  }
+};
+
+const asyncValidate = (values, dispatch, props, blurredField) => {
```

Comment:
Why is the asyncValidate used here? It doesn't looks like anything is validated asynchronously here?

Reply:
AsyncValidate is used to round the amount field after the focus is lost.

Reply:
@DGB-UNAM-FOLIO please use `onBlur` directly on the amount input field.

Reply:
I was thinking the same thing, @mkuklis. Validation functions should not be used to implement side-effects.

Reply:
We had used onBlur, but it didn't work, because of this we handled it this way.

Reply:
Could you describe in more details what didn't work? Did `onBlur` not executed?

Reply:
We use ``onBlur``, but it does not perform the operations as we expected.
At first it worked, but one day it stopped working well, so we had to find another way to make it work.